### PR TITLE
Remove duplicate transactions from input CSV

### PR DESCRIPTION
**Pull request by arkadasfynarfin**
## Description

For a given input CSV file, we need to remove duplicate transactions based on payee identifier type, payee identifier, amount and currency.

- If ordering is enabled and the transactions are order, then we compare transactions next to each other for duplicates.
- If ordering is disabled, then we use a set to find out duplicates.

For all duplicate records, we are setting the note to `Duplicate transaction.` so that it can later be failed.

Jira link - [GOV-325](https://mifosforge.jira.com/browse/GOV-325)

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [ ] Design related bullet points or design document link related to this PR added in the description above. 

- [ ] Updated corresponding Postman Collection or Api documentation for the changes in this PR.

- [ ] Create/update unit or integration tests for verifying the changes made.

- [ ] Add required Swagger annotation and update API documentation with details of any API changes if applicable

- [ ] Followed the naming conventions as given in https://docs.google.com/document/d/1Q4vaMSzrTxxh9TS0RILuNkSkYCxotuYk1Xe0CMIkkCU/edit?usp=sharing

[GOV-325]: https://mifosforge.jira.com/browse/GOV-325?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by arkadasfynarfin** [comment]
@fynmanoj Please review this PR.
cc: @avikganguly01

**Review by fynmanoj — CHANGES_REQUESTED** [review]

File: src/main/java/org/mifos/processor/bulk/zeebe/worker/OrderingWorker.java [hunk 1]
```diff
@@ -26,26 +28,85 @@ public void setup() {
...
+            return;
+        }
+
+        for(Transaction transaction : transactionList){
+            String payeeDetail = fetchPayeeDetail(transaction);
+            if(set.contains(payeeDetail)){
+                transaction.setNote("Duplicate transaction.");
+            }
+            else{
+                set.add(payeeDetail);
```
Comment (fynmanoj) [inline src/main/java/org/mifos/processor/bulk/zeebe/worker/OrderingWorker.java:100]: warning. adding all transactions to a set can cause memory issues. address this in another ticket

File: src/main/java/org/mifos/processor/bulk/zeebe/worker/OrderingWorker.java [hunk 2]
```diff
@@ -26,26 +28,85 @@ public void setup() {
...
+
+    private String fetchPayeeDetail(Transaction transaction){
+        if(Objects.isNull(transaction)){
+            return null;
+        }
+        String payeeIdentifier = transaction.getPayeeIdentifier();
+        String payeeIdentifierType = transaction.getPayeeIdentifierType();
+        String amount = transaction.getAmount();
+        String currency = transaction.getCurrency();
+        return payeeIdentifierType + payeeIdentifier + amount + currency;
```
Comment (fynmanoj) [inline src/main/java/org/mifos/processor/bulk/zeebe/worker/OrderingWorker.java:113]: concatenating with null check can cause null pointer issues.

**Review by fynmanoj — CHANGES_REQUESTED** [review]

File: src/main/java/org/mifos/processor/bulk/zeebe/worker/OrderingWorker.java (hunk 2 above)
Comment (fynmanoj) [reply src/main/java/org/mifos/processor/bulk/zeebe/worker/OrderingWorker.java:113]: String.format("%s%s%s%s",payeeIdentifierType ,payeeIdentifier , amount ,currency)

File: src/main/java/org/mifos/processor/bulk/zeebe/worker/OrderingWorker.java [hunk 3]
```diff
@@ -26,26 +32,85 @@ public void setup() {
...
+            Transaction currentTransaction = transactionList.get(i);
+            Transaction nextTransaction = transactionList.get(i + 1);
+
+            if (currentTransaction == null || nextTransaction == null) {
+                continue;
+            }
+            String currentPayeeDetail = fetchPayeeDetail(currentTransaction);
+            String nextPayeeDetail = fetchPayeeDetail(nextTransaction);
+
+            if (currentPayeeDetail.equals(nextPayeeDetail)) {
```
Comment (fynmanoj) [inline src/main/java/org/mifos/processor/bulk/zeebe/worker/OrderingWorker.java:81]: not null safe , this method: fetchPayeeDetail(currentTransaction) can possibly return null ( as the method has a return null )

File: src/main/java/org/mifos/processor/bulk/zeebe/worker/OrderingWorker.java [hunk 4]
```diff
@@ -26,26 +32,85 @@ public void setup() {
...
+                transaction.setNote("Duplicate transaction.");
+            }
+            else{
+                set.add(payeeDetail);
+            }
+        }
+    }
+
+    private String fetchPayeeDetail(Transaction transaction){
+        if(Objects.isNull(transaction)){
```
Comment (fynmanoj) [inline src/main/java/org/mifos/processor/bulk/zeebe/worker/OrderingWorker.java:106]: null check already exist before this call. unless this method is reused, you can remove the redundant code.

**Review by fynmanoj — CHANGES_REQUESTED** [review]

**Pull request merged**